src/passes/NameTypes.cpp

```diff
@@ -34,13 +34,36 @@ struct NameTypes : public Pass {
     // Find all the types.
     std::vector<HeapType> types = ModuleUtils::collectHeapTypes(*module);
 
+    std::unordered_set<Name> used;
+
     // Ensure simple names. If a name already exists, and is short enough, keep
     // it.
     size_t i = 0;
     for (auto& type : types) {
       if (module->typeNames.count(type) == 0 ||
           module->typeNames[type].name.size() >= NameLenLimit) {
-        module->typeNames[type].name = "type$" + std::to_string(i++);
+        module->typeNames[type].name = "type_" + std::to_string(i++);
```

Comment:
I guess `type_0` will be linted to `type`. Is it worth special-casing "type" to avoid that?

Reply:
I don't see why? I feel like I'm not seeing what you find wrong with that linting...

Reply:
I guess it's a more general QoL issue: if there is `somename_X` and `somename_Y`, there's not a principled way to determine which one deserves to be improved to `somename`. I would have a slight preference for improving neither in that case, but improving one arbitrarily also seems reasonable.

Reply:
I see now, thanks. Yeah, maybe in some cases it might be better to avoid linting either name. I think in most cases however it will be worth it. Anyhow, if there's feedback on this we can iterate more later.